tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,452 @@
+                        konnected.CONF_DEVICES: [{konnected.CONF_ID: "aabbccddeeff"}],
+                    }
+                },
+            )
+            is True
+        )
+
+    assert len(mock_panel.mock_calls) == 2
+    p_hass, p_entry = mock_panel.mock_calls[0][1]
+
+    assert p_hass is hass
+    assert p_entry is entry
+
+
+async def test_unload_entry(hass):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=konnected.DOMAIN, data={konnected.CONF_ID: "aabbccddeeff"}
+    )
+    entry.add_to_hass(hass)
+
+    with patch.object(konnected, "AlarmPanel") as mock_panel:
+
+        def mock_constructor(hass, entry):
+            """Fake the panel constructor."""
+            return mock_panel
+
+        def save_data():
+            hass.data[konnected.DOMAIN]["devices"]["aabbccddeeff"] = {"some": "thing"}
+
+        mock_panel.side_effect = mock_constructor
+        mock_panel.async_save_data.side_effect = save_data
+        mock_panel.async_save_data.return_value = mock_coro()
+        mock_panel.async_connect.return_value = mock_coro()
+        assert await async_setup_component(hass, konnected.DOMAIN, {}) is True
+
+    assert hass.data[konnected.DOMAIN]["devices"].get("aabbccddeeff") is not None
+
+    assert await konnected.async_unload_entry(hass, entry)
+    assert hass.data[konnected.DOMAIN]["devices"] == {}
+
+
+async def test_api(hass, aiohttp_client):
+    """Test callback view."""
+    await async_setup_component(hass, "http", {"http": {}})
+
+    device_config = config_flow.DEVICE_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "binary_sensors": [
+                {"zone": "1", "type": "door"},
+                {"zone": "2", "type": "window", "name": "winder", "inverse": True},
+                {"zone": "3", "type": "door"},
+            ],
+            "sensors": [
+                {"zone": "4", "type": "dht"},
+                {"zone": "5", "type": "ds18b20", "name": "temper"},
+            ],
+            "switches": [
+                {
+                    "zone": "out",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+                {"zone": "6"},
+            ],
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected", title="Konnected Alarm Panel", data=device_config
+    )
+    entry.add_to_hass(hass)
+
+    with patch("konnected.Client") as mock_panel:
+
+        def mock_constructor(host, port, websession):
+            """Fake the panel constructor."""
+            mock_panel.host = host
+            mock_panel.port = port
+            return mock_panel
+
+        mock_panel.side_effect = mock_constructor
+        mock_panel.get_status.return_value = mock_coro(
```

Comment:
After switching to `asynctest.patch` we don't need to use `mock_coro` anymore. `mock_coro` is legacy. Just set the return value of the patched coroutine as normal.